docs/tools/offline/toolchain_profiles.md

```diff
@@ -75,10 +75,15 @@ And so on.
 
 ### API perspective
 
-The toolchains take in an optional argument, ``build_profile``, that will contain a map from flag types to lists of flags. This looks exactly the same in Python as it does in the JSON format above.
+The toolchains take in an optional argument, ``build_profile``, that maps from flag types to lists of flags. When provided, this argument must contain a dict mapping from flag types to a list of flags to provide to the compiler. If this argument is not provided, a build profile is used that contains no flags in each required flag type. The required flag types are:
```

Comment:
Query: Who or what provides the argument? Also, what uses the build profile? The compiler?

Reply:
> Who or what provides the argument?

It's an API description, that's out of scope.

> Also, what uses the build profile?

The toolschains, they take it in.... I think there's a disconnect here.